test/test_issue1447.py

```diff
@@ -0,0 +1,140 @@
+        example_BlankNodeOrIRI
+    ] = {kb_https_uriref}
+    computed_nodes_using_predicate_q: typing.Set[
+        example_BlankNodeOrIRI
+    ] = set()
+    for triple in graph.triples((None, predicate_q, None)):
+        computed_nodes_using_predicate_q.add(triple[0])
+    assert expected_nodes_using_predicate_q == computed_nodes_using_predicate_q
+
+    one_usage_query = """\
+SELECT ?resource
+WHERE {
+  ?resource <http://example.org/predicates#p> 1 .
+}
+"""
+
+    expected_one_usage: typing.Set[example_BlankNodeOrIRI] = {
+        kb_bnode,
+        kb_http_uriref,
+        kb_https_uriref,
+        kb_urn_uriref,
+    }
+    computed_one_usage: typing.Set[example_BlankNodeOrIRI] = set()
+    for one_usage_result in graph.query(one_usage_query):
+        computed_one_usage.add(one_usage_result[0])
+    assert expected_one_usage == computed_one_usage
+
+    # The purpose of this query is confirming the believed return
+    # signature of graph.query() is determined at *each* call of
+    # graph.query(), rather than the first.  I.e. there should not be a
+    # type error when the first call returns a length-one result tuple,
+    # and the second a length-two result tuple.
+    two_usage_query = """\
+SELECT ?resource ?predicate
+WHERE {
+  ?resource ?predicate "2"^^xsd:integer .
+}
+"""
+
+    expected_two_usage: typing.Set[
+        typing.Tuple[
+            example_BlankNodeOrIRI,
+            example_BlankNodeOrIRI,
+        ]
+    ] = {(kb_https_uriref, predicate_p), (kb_https_uriref, predicate_q)}
+    computed_two_usage: typing.Set[
+        typing.Tuple[
+            example_BlankNodeOrIRI,
+            example_BlankNodeOrIRI,
+        ]
+    ] = set()
+    for two_usage_result in graph.query(two_usage_query):
+        computed_two_usage.add(two_usage_result)
+    assert expected_two_usage == computed_two_usage
+
+    nsdict = {k: v for (k, v) in graph.namespace_manager.namespaces()}
+
+    prepared_one_usage_query = rdflib.plugins.sparql.processor.prepareQuery(
+        one_usage_query, initNs=nsdict
+    )
+    computed_one_usage_from_prepared_query: typing.Set[
+        example_BlankNodeOrIRI
+    ] = set()
+    for prepared_one_usage_result in graph.query(prepared_one_usage_query):
+        computed_one_usage_from_prepared_query.add(prepared_one_usage_result[0])
+    assert expected_one_usage == computed_one_usage_from_prepared_query
+
+    for node_using_one in sorted(computed_one_usage):
+        graph.add((node_using_one, predicate_r, literal_true))
+
+    python_one: int = literal_one.toPython()
+    assert python_one == 1
+
+    python_two: int = literal_two.toPython()
+    assert python_two == 2
+
+    python_true: bool = literal_true.toPython()
+    assert python_true == True
+
+    python_iri: str = kb_https_uriref.toPython()
+    assert python_iri == "https://example.org/kb/y"
+
+
+def main() -> None:
+    test_rdflib_query_exercise()
+
+
+if __name__ == "__main__":
+    main()
```

Comment:
I think it is best to not add this, if pytest features are used this likely may not work, better to just run:

```bash
python -m pytest test/test_issue1447.py
```

Reply:
Thanks, I wasn't clear on what the transition status was from `unittest` to `pytest`.  The recent doc updates help clarify the matter too.  Removing.